x-pack/solutions/search/plugins/search_synonyms/server/lib/fetch_synonym_rule.ts

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ElasticsearchClient } from '@kbn/core/server';
+
+export const fetchSynonymRule = async (
+  client: ElasticsearchClient,
+  synonymsSetId: string,
+  ruleId: string
+) => {
+  const result = await client.synonyms.getSynonymRule({
+    set_id: synonymsSetId,
+    rule_id: ruleId,
+  });
+  return result;
```

Comment:
If we're not handling errors and just immediately returning the results then we can just return the promise.

```suggestion
  return client.synonyms.getSynonymRule({
    set_id: synonymsSetId,
    rule_id: ruleId,
  });
```